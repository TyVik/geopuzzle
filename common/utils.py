import string

from django.utils.crypto import get_random_string


def random_string():
    return get_random_string(12, string.ascii_lowercase + string.digits)
