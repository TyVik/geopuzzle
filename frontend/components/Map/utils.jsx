/* global google */


function decodePolygon(polygon) {
  return polygon.map(polygon => (google.maps.geometry.encoding.decodePath(polygon)));
}

export { decodePolygon };
