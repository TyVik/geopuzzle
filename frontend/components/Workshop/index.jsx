'use strict';
import React from "react";
import InfiniteScroll from 'react-infinite-scroll-component';


class Workshop extends React.Component {
    constructor(props) {
        super(props);
        this.state = {puzzles: [], page: 0};
    }

    componentDidMount() {
        this.fetchPage(1);
    }

    fetchPage(page) {
        fetch(`${location.pathname}items/?page=${page}`, {method: 'GET'})
            .then(response => response.json())
            .then(data => {
                let items = this.state.puzzles.concat(data);
                this.setState({...this.state, puzzles: items, page: page});
            });
    }

    fetchNextPage = () => {
        this.fetchPage(this.state.page + 1);
    };

    render() {
        if (this.state.page === 0) {
            return <h4>Loading...</h4>;
        }
        let puzzles = this.state.puzzles;
        return <InfiniteScroll dataLength={puzzles.length} next={this.fetchNextPage}
                               hasMore={puzzles.length < this.props.count}
                               loader={<h4>Loading...</h4>} children={puzzles}>
            {puzzles.map(puzzle =>
                <div className="col-md-3 col-sm-4 col-xs-6 item-container" key={puzzle.url}>
                    <a href={puzzle.url}>
                      <img className="img-responsive img-rounded" src={puzzle.image}
                           alt={puzzle.name}/>
                    </a>
                    <div className="text-center">{puzzle.name}</div>
                </div>)}
        </InfiniteScroll>;
    }
}

export default Workshop;
