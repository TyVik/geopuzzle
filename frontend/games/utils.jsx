import messages from "../locale/messages";

const prepareInfobox = (json) => {
  if (json.area) {
    json.area = `${Number(json.area).toLocaleString()} ${messages[window.__LANGUAGE__].km2}`;
  }
  if (json.population) {
    json.population = Number(json.population).toLocaleString();
  }
  ['depth', 'length', 'width', 'height'].forEach(item => {
    if (json[item]) {
      json[item] = `${Number(json[item]).toLocaleString()} ${messages[window.__LANGUAGE__].m}`;
    }
  });
  return json;
};

function shuffle(a) {
  for (let i = a.length; i; i--) {
    let j = Math.floor(Math.random() * i);
    [a[i - 1], a[j]] = [a[j], a[i - 1]];
  }
  return a;
}


export {prepareInfobox, shuffle};