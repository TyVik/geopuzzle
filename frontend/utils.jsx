'use strict';
import 'whatwg-fetch';
import Cookies from 'js-cookie';

import localization from "./localization";

function moveTo(paths, from, to) {
    let newPoints =[];
    let newPaths = [];

    for (let p = 0; p < paths.length; p++) {
        let path = paths[p];
        newPoints.push([]);

        for (let i = 0; i < path.length; i++) {
            newPoints[newPoints.length-1].push({
                heading: google.maps.geometry.spherical.computeHeading(from, path[i]),
                distance: google.maps.geometry.spherical.computeDistanceBetween(from, path[i])
            });
        }
    }

    for (let j = 0, jl = newPoints.length; j < jl; j++) {
        let shapeCoords = [],
            relativePoint = newPoints[j];
        for (let k = 0, kl = relativePoint.length; k < kl; k++) {
            shapeCoords.push(google.maps.geometry.spherical.computeOffset(
                to,
                relativePoint[k].distance,
                relativePoint[k].heading
            ));
        }
        newPaths.push(shapeCoords);
    }
    return newPaths;
}


function decodePolygon(polygon) {
    return polygon.map(polygon => (google.maps.geometry.encoding.decodePath(polygon)));
}

const prepareInfobox = (json) => {
    if (json.area) {
        json.area = Number(json.area).toLocaleString() + ' ' + localization.km2;
    }
    if (json.population) {
        json.population = Number(json.population).toLocaleString();
    }
    return json;
};

function shuffle(a) {
    for (let i = a.length; i; i--) {
        let j = Math.floor(Math.random() * i);
        [a[i - 1], a[j]] = [a[j], a[i - 1]];
    }
    return a;
}


const CSRFfetch = (url, options) => {
    let headers = options.headers || new Headers();
    headers.append('X-CSRFTOKEN', Cookies.get('csrftoken'));
    return fetch(url, {...options, headers: headers, credentials: 'same-origin'});
};


export {moveTo, decodePolygon, prepareInfobox, shuffle, CSRFfetch};
