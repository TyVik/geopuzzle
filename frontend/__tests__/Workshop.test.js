'use strict';
import React from "react";
import Workshop from "../workshop/index";
import {createComponentWithIntl} from "./utils";


describe('shallow <Workshop /> components', () => {
  it('render', () => {
    let order = [
      ["title_asc", "Title \u2193"], ["title_desc", "Title \u2191"],
      ["created_asc", "Created \u2193"], ["created_desc", "Created \u2191"]
    ];
    expect(createComponentWithIntl(<Workshop orderOptions={order}/>)).toMatchSnapshot('Workshop');
  });
});
