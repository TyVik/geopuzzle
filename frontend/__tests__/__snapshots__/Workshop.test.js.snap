// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`shallow <Workshop /> components render: Workshop 1`] = `
Array [
  <form
    class="row justify-content-between"
  >
    <div
      class="form-group col-md-7 col-sm-12"
    >
      <label
        for="search-label"
      >
        <span>
          Search by title
        </span>
        :
      </label>
      <input
        class="form-control"
        id="search-input"
        maxlength="50"
        type="text"
        value=""
      />
    </div>
    <div
      class="form-group col-md-5 col-sm-12"
    >
      <label
        for="order-label"
      >
        <span>
          Order by
        </span>
        :
      </label>
      <div
        class=" css-2b097c-container"
      >
        <div
          class=" css-yk16xz-control"
        >
          <div
            class=" css-1hwfws3"
          >
            <div
              class=" css-1wa3eu0-placeholder"
            >
              Select...
            </div>
            <div
              class="css-1g6gooi"
            >
              <div
                class=""
                style="display:inline-block"
              >
                <input
                  aria-autocomplete="list"
                  autocapitalize="none"
                  autocomplete="off"
                  autocorrect="off"
                  id="react-select-2-input"
                  spellcheck="false"
                  style="box-sizing:content-box;width:1px;label:input;background:0;border:0;font-size:inherit;opacity:1;outline:0;padding:0;color:inherit"
                  tabindex="0"
                  type="text"
                  value=""
                />
                <div
                  style="position:absolute;top:0;left:0;visibility:hidden;height:0;overflow:scroll;white-space:pre"
                />
              </div>
            </div>
          </div>
          <div
            class=" css-1wy0on6"
          >
            <span
              class=" css-1okebmr-indicatorSeparator"
            />
            <div
              aria-hidden="true"
              class=" css-tlfecz-indicatorContainer"
            >
              <svg
                aria-hidden="true"
                class="css-19bqh2r"
                focusable="false"
                height="20"
                viewBox="0 0 20 20"
                width="20"
              >
                <path
                  d="M4.516 7.548c0.436-0.446 1.043-0.481 1.576 0l3.908 3.747 3.908-3.747c0.533-0.481 1.141-0.446 1.574 0 0.436 0.445 0.408 1.197 0 1.615-0.406 0.418-4.695 4.502-4.695 4.502-0.217 0.223-0.502 0.335-0.787 0.335s-0.57-0.112-0.789-0.335c0 0-4.287-4.084-4.695-4.502s-0.436-1.17 0-1.615z"
                />
              </svg>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="form-group col-md-6 col-sm-12"
    >
      <label
        for="search-label"
      >
        <span>
          Filter by tag
        </span>
        :
      </label>
      <div
        class=" css-2b097c-container"
      >
        <div
          class=" css-yk16xz-control"
        >
          <div
            class=" css-1hwfws3"
          >
            <div
              class=" css-1wa3eu0-placeholder"
            >
              Select...
            </div>
            <div
              class="css-1g6gooi"
            >
              <div
                class=""
                style="display:inline-block"
              >
                <input
                  aria-autocomplete="list"
                  autocapitalize="none"
                  autocomplete="off"
                  autocorrect="off"
                  id="react-select-3-input"
                  spellcheck="false"
                  style="box-sizing:content-box;width:1px;label:input;background:0;border:0;font-size:inherit;opacity:1;outline:0;padding:0;color:inherit"
                  tabindex="0"
                  type="text"
                  value=""
                />
                <div
                  style="position:absolute;top:0;left:0;visibility:hidden;height:0;overflow:scroll;white-space:pre"
                />
              </div>
            </div>
          </div>
          <div
            class=" css-1wy0on6"
          >
            <div
              aria-hidden="true"
              class=" css-at12u2-loadingIndicator"
            >
              <span
                class="css-d0ptw1"
              />
              <span
                class="css-3utax6"
              />
              <span
                class="css-1qyc6ze"
              />
            </div>
            <span
              class=" css-1okebmr-indicatorSeparator"
            />
            <div
              aria-hidden="true"
              class=" css-tlfecz-indicatorContainer"
            >
              <svg
                aria-hidden="true"
                class="css-19bqh2r"
                focusable="false"
                height="20"
                viewBox="0 0 20 20"
                width="20"
              >
                <path
                  d="M4.516 7.548c0.436-0.446 1.043-0.481 1.576 0l3.908 3.747 3.908-3.747c0.533-0.481 1.141-0.446 1.574 0 0.436 0.445 0.408 1.197 0 1.615-0.406 0.418-4.695 4.502-4.695 4.502-0.217 0.223-0.502 0.335-0.787 0.335s-0.57-0.112-0.789-0.335c0 0-4.287-4.084-4.695-4.502s-0.436-1.17 0-1.615z"
                />
              </svg>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="form-group col-md-6 col-sm-12"
    >
      <label
        for="search-label"
      >
        <span>
          Filter by user
        </span>
        :
      </label>
      <div
        class=" css-2b097c-container"
      >
        <div
          class=" css-yk16xz-control"
        >
          <div
            class=" css-1hwfws3"
          >
            <div
              class=" css-1wa3eu0-placeholder"
            >
              Select...
            </div>
            <div
              class="css-1g6gooi"
            >
              <div
                class=""
                style="display:inline-block"
              >
                <input
                  aria-autocomplete="list"
                  autocapitalize="none"
                  autocomplete="off"
                  autocorrect="off"
                  id="react-select-4-input"
                  spellcheck="false"
                  style="box-sizing:content-box;width:1px;label:input;background:0;border:0;font-size:inherit;opacity:1;outline:0;padding:0;color:inherit"
                  tabindex="0"
                  type="text"
                  value=""
                />
                <div
                  style="position:absolute;top:0;left:0;visibility:hidden;height:0;overflow:scroll;white-space:pre"
                />
              </div>
            </div>
          </div>
          <div
            class=" css-1wy0on6"
          >
            <div
              aria-hidden="true"
              class=" css-at12u2-loadingIndicator"
            >
              <span
                class="css-d0ptw1"
              />
              <span
                class="css-3utax6"
              />
              <span
                class="css-1qyc6ze"
              />
            </div>
            <span
              class=" css-1okebmr-indicatorSeparator"
            />
            <div
              aria-hidden="true"
              class=" css-tlfecz-indicatorContainer"
            >
              <svg
                aria-hidden="true"
                class="css-19bqh2r"
                focusable="false"
                height="20"
                viewBox="0 0 20 20"
                width="20"
              >
                <path
                  d="M4.516 7.548c0.436-0.446 1.043-0.481 1.576 0l3.908 3.747 3.908-3.747c0.533-0.481 1.141-0.446 1.574 0 0.436 0.445 0.408 1.197 0 1.615-0.406 0.418-4.695 4.502-4.695 4.502-0.217 0.223-0.502 0.335-0.787 0.335s-0.57-0.112-0.789-0.335c0 0-4.287-4.084-4.695-4.502s-0.436-1.17 0-1.615z"
                />
              </svg>
            </div>
          </div>
        </div>
      </div>
    </div>
  </form>,
  <div>
    <div
      class="infinite-scroll-component row"
      style="height:auto;overflow:auto;-webkit-overflow-scrolling:touch"
    >
      <div
        class="loading_wrapper"
      >
        <h2
          class="loading"
        >
          <span>
            Loading...
          </span>
        </h2>
      </div>
    </div>
  </div>,
]
`;
